server/index.js

```diff
@@ -2,9 +2,33 @@ require('dotenv/config');
 const express = require('express');
 const errorMiddleware = require('./error-middleware');
 const staticMiddleware = require('./static-middleware');
+const ClientError = require('./client-error');
 
 const app = express();
 
+const petfinder = require('@petfinder/petfinder-js');
+const client = new petfinder.Client({ apiKey: process.env.PETFINDER_KEY, secret: process.env.PETFINDER_SECRET });
+
+app.get('/api/pets/:location/:type', (req, res) => {
+  const { location } = req.params;
+  const { type } = req.params;
+  if (!location || !type) {
+    throw new ClientError(400, 'Invalid location or type selection');
+  }
+  client.animal.search({
+    limit: 1,
+    distance: 20,
+    sort: 'random',
+    location: location,
+    type: type
+  })
+    .then(response => {
+      const animal = response.data.animals;
+      res.status(200).json(animal);
```

Comment:
If you only want to send one animal, then sending the object (response.data.animals[0]) instead of the array holding the object, is preferable.